### ISSUE TITLE
E575: viminfo: Illegal starting char in line

### ISSUE THREAD

**Issue by mgedmin** [vim/vim#870]
Lastest vim (7.4.1941) produces this error on startup and also right before exiting:

```
E575: viminfo: Illegal starting char in line: ^Ihi</i>^[llx
Press ENTER or type command to continue
```

I think this is a bug since vim itself is the program responsible for writing to my .viminfo.

Looking at the .viminfo itself, here's what I see:

```
"o  CHAR    0
    hi</i>llx
|3,0,24,0,1,0,0,"hi</i>llx"
```

There's an ESC character in front of 'llx', in case it doesn't make it through email.  Also, there are no spaces on those three lines, just tabs.

In case the problem is on a previous line that somehow makes vim lose sync, here's what comes before:

```
"m  CHAR    0
    807
<mgggg€kd€kd€kd€kd€kd€kd€ku€ku€ku€ku*'''.€kr€klbidatetime.€@7€kl(2, 0):w
€kr€kd€kd€kd€kd€kd€kd€kd€kd€kd€ku€ku€ku€ku€ku€ku€ku€klodd€ku€kr€kd€kN€kN€kN€kN€klo
tiemlog.window_for_dat(€kb€kby(date€kb€kb€kbda€kb€kbatetime.date('€kb2014, 11, 12))€kh€kr€kr€kr€kr€kr€kr€kr€kr€kr€kr€kr€kb€kre€kh[1;5Cw = €kr€kdgg/TimeWindow
€kN€kN€k3€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kloself.asseert€kb€kb€kbrtEqual(list(w.all_entries), [])o
timeo€kblog.add_entry€kr/add_ent
/add_e/add€kb€kbppend_en
/€kb€kuq/class Tim
€k3zMzr€kN€kl€kl€kl€kl€kl€kl€kl€klCppend('started **')€kr€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€ku€kl€kr€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€ku€ku€ku€ku€ku€ku€ku€ku€ku€kd€kd€żUi, now=datetime.datetime(2014, 11, 12, 10, 00€@7)%
|3,0,22,0,1,0,0,>806
|<"mgggg€kd€kd€kd€kd€kd€kd€ku€ku€ku€ku*'''.€kr€klbidatetime.€@7€kl(2, 0):w
€kr€kd€kd€kd€kd€kd€kd€kd€kd€kd€ku€ku€ku€ku€ku€ku€ku€klodd€ku€kr€kd€kN€kN€kN€kN€klo
tiemlog.window_for_dat(€kb€kby(date€kb€kb€kbda€kb€kbatetime.date('€kb2014, 11, 12))€kh€kr€kr€kr€kr€kr€kr€kr€kr€kr€kr€kr€kb€kre€kh[1;5Cw = €kr€kdgg/TimeWindow
€kN€kN€k3€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kN€kloself.asseert€kb€kb€kbrtEqual(list(w.all_entries), [])o
timeo€kblog.add_entry€kr/add_ent
|</add_e/add€kb€kbppend_en
/€kb€kuq/class Tim
€k3zMzr€kN€kl€kl€kl€kl€kl€kl€kl€klCppend('started **')€kr€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€ku€kl€kr€kd€kd€kd€kd€kd€kd€kd€kd€kd€kd€ku€ku€ku€ku€ku€ku€ku€ku€ku€kd€kd€żUi, now=datetime.datetime(2014, 11, 12, 10, 00€@7)%"
```

This looks like a macro, possibly recorded by accident by accidentally hitting q.

When I run vim, register "o looks just like in viminfo (`"hi</i><esc>llx"`), while register "m is blank.

vim 1.689 doesn't complain and it can load register "m fine.

**Comment by mgedmin** [comment]
~~git bisect blames commit a890f5e34887bff7616bdb4b9ee0bf98c8d2a8f0 for this~~

EDIT: never mind, I messed up git bisect somehow.

**Comment by mgedmin** [comment]
I don't see this error with 7.4.1918, but I see this with [v7.4.1919](http://github.com/vim/vim/commit/46bbb0c4ba27395859dfeaa26938483946bb4ec2).

**Comment by brammool** [comment]
Both github and mail mess up the text.  Can you please email those lines in an attachment, so that they arrive exactly as they were?  It's important to get the exact text, I tried several texts with weird characters and they all work fine.

You can also try deleting one line at a time to check that it's the one causing trouble.  Do include continuation lines then (starting with "|<" or with Tab or with "<").

**Comment by mgedmin** [comment]
I can no longer reproduce this with version 7.4.1944.

For extra fun: the copy of the .viminfo I made so I could reproduce this bug later doesn't look like what I described in my original report.  All the lines starting with a `|` have been moved into one location, preceded by a comment saying "Bar lines, copied verbatim".  I should've made a copy of the .viminfo before I started playing with git bisecting, not after.

**Comment by mgedmin** [comment]
I found a different copy of a .viminfo, and I can reproduce this again, with 7.4.1944.

I've uploaded a trimmed down version of the .viminfo to https://gist.github.com/mgedmin/4ba39383f51efdea64f5e4e40d4cfd5c, and I'll send a copy to Bram by email.

**Comment by brammool** [comment]
Marius Gedminas wrote:

> I can no longer reproduce this with version 7.4.1944.

Glad to hear that.

> For extra fun: the copy of the .viminfo I made so I could reproduce
> this bug later doesn't look like what I described in my original
> report.  All the lines starting with a `|` have been moved into one
> location, preceded by a comment saying "Bar lines, copied verbatim".
> I should've made a copy of the .viminfo before I started playing with
> git bisecting, not after.

Yeah, editing .viminfo with Vim is tricky, since edits you make to the
file will be merged when exiting Vim.  Making 'viminfo' empty before
exiting helps.

The idea with the barlines is that when running both an older and a
newer version of Vim, most history will still work.  Copying the
barlines was added months ago for that reason.

Keep an eye out for the merging to not work properly, it's quite tricky
to get right.

There are still a couple more things where merging based on a timestamp
is useful.  I'll implement it when I have time.

## 

Two percent of zero is almost nothing.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by mgedmin** [comment]
Fixed in ecefe71704850b94df44f65fc756c1551ec68388 (7.4.1947).

**Comment by gamesharkmike** [comment]
I got this message when I was out of disk space, the file must have been corrupted. I noticed an entry not starting with `>` so I deleted that section. The file seems to have been regenerated because it is different. Either way the error doesn't show up whenever I start vim. Hope this post is helpful for anyone who is getting this error who has no disk space left.
